java/code/src/com/redhat/rhn/frontend/action/channel/ssm/BaseSubscribeAction.java

```diff
@@ -694,16 +694,15 @@ else if (toId != null && toId == -1L) {
 
             // Skipped messages
             srvrs = skipped.get(toId);
-            if (srvrs.isEmpty()) {
-                continue;
-            }
-            else if (toId != null && toId == -1L) {
-                am = new ActionMessage("basesub.jsp.skip-default", srvrs.size());
-                msgs.add(ActionMessages.GLOBAL_MESSAGE, am);
-            }
-            else {
-                am = new ActionMessage("basesub.jsp.skip", srvrs.size(), c.getName());
-                msgs.add(ActionMessages.GLOBAL_MESSAGE, am);
+            if (!srvrs.isEmpty()) {
+                if (toId != null && toId == -1L) {
+                    am = new ActionMessage("basesub.jsp.skip-default", srvrs.size());
+                    msgs.add(ActionMessages.GLOBAL_MESSAGE, am);
+                }
+                else {
+                    am = new ActionMessage("basesub.jsp.skip", srvrs.size(), c.getName());
+                    msgs.add(ActionMessages.GLOBAL_MESSAGE, am);
+                }
```

Comment:
Same here, I would just remove the redundant `else` rather then inverting the `if`. You can also do the same for the `if` block that precedes this one. It should also fix the cognitive complexity issue.